features/netsocket/RAWSocket.h

```diff
@@ -0,0 +1,175 @@
+     *  @param size     Size of the buffer in bytes.
+     *  @return         Number of received bytes on success, negative error
+     *                  code on failure.
+     */
+    nsapi_size_or_error_t recvfrom(SocketAddress *address,
+                                   void *data, nsapi_size_t size);
+
+    /** Set the remote address for next send() call and filtering
+     *  of incoming packets. To reset the address, zero initialized
+     *  SocketAddress must be in the address parameter.
+     *
+     *  @param address  The SocketAddress of the remote host.
+     *  @return         0 on success, negative error code on failure.
+     */
+    nsapi_error_t connect(const SocketAddress &address);
+
+    /** Send a datagram to connected remote address.
+     *
+     *  By default, send blocks until all data is sent. If socket is set to
+     *  nonblocking or times out, a partial amount can be written.
+     *  NSAPI_ERROR_WOULD_BLOCK is returned if no data was written.
+     *
+     *  @note The socket must be connected to a remote host before send() call.
+     *
+     *  @param data     Buffer of data to send to the host.
+     *  @param size     Size of the buffer in bytes.
+     *  @return         Number of sent bytes on success, negative error
+     *                  code on failure.
+     */
+    nsapi_size_or_error_t send(const void *data, nsapi_size_t size);
+
+    /** Receive data from a socket.
+     *
+     *  This is equivalent to calling recvfrom(NULL, data, size).
+     *
+     *  If the socket is connected, only packets coming from a connected peer address
+     *  are accepted.
+     *
+     *  By default, recv blocks until some data is received. If socket is set to
+     *  nonblocking or times out, NSAPI_ERROR_WOULD_BLOCK can be returned to
+     *  indicate no data.
+     *
+     *  @note recv() is allowed write to data buffer even if error occurs.
+     *
+     *  @param data     Pointer to buffer for data received from the host.
+     *  @param size     Size of the buffer in bytes.
+     *  @return         Number of received bytes on success, negative error
+     *                  code on failure.
+     */
+    nsapi_size_or_error_t recv(void *data, nsapi_size_t size);
+
+    /** Not implemented for RAWSocket.
+     *
+     *  @param error      Not used.
+     *  @return           NSAPI_ERROR_UNSUPPORTED
+     */
+    Socket *accept(nsapi_error_t *error = NULL);
+
+    /** Not implemented for RAWSocket.
+     *
+     *  @param backlog    Not used.
+     *  @return           NSAPI_ERROR_UNSUPPORTED
+     */
+    nsapi_error_t listen(int backlog = 1);
+
+#if !defined(DOXYGEN_ONLY)
+
+protected:
+    nsapi_protocol_t get_proto();
+
+private:
+    /** Create a socket out of a given socket
+     *
+     *  To be used within accept() function. Close() will clean this up.
+     */
+    RAWSocket(RAWSocket *parent, nsapi_socket_t socket, SocketAddress address);
```

Comment:
This was copied from TCPSocket? Not needed for raw.

Reply:
I made this change by copying from UDPSocket.cpp, yes it is not needed for RAWSocket